vm/actor/src/builtin/miner/types.rs

```diff
@@ -117,51 +191,85 @@ pub struct ReportConsensusFaultParams {
     pub effective_at: ChainEpoch,
 }
+
 #[derive(Debug, PartialEq, Clone, Serialize_tuple, Deserialize_tuple)]
 pub struct SectorPreCommitInfo {
-    pub registered_proof: RegisteredSealProof,
+    pub seal_proof: RegisteredSealProof,
     pub sector_number: SectorNumber,
     /// CommR
     pub sealed_cid: Cid,
     pub seal_rand_epoch: ChainEpoch,
     pub deal_ids: Vec<DealID>,
-    /// Sector Expiration
     pub expiration: ChainEpoch,
+    /// Whether to replace a "committed capacity" no-deal sector (requires non-empty DealIDs)
+    pub replace_capacity: bool,
+    /// The committed capacity sector to replace, and its deadline/partition location
+    pub replace_sector_deadline: u64,
+    pub replace_sector_partition: u64,
+    pub replace_sector_number: SectorNumber,
 }
+
+/// Information stored on-chain for a pre-committed sector.
 #[derive(Debug, PartialEq, Clone, Serialize_tuple, Deserialize_tuple)]
 pub struct SectorPreCommitOnChainInfo {
     pub info: SectorPreCommitInfo,
     #[serde(with = "bigint_ser")]
     pub pre_commit_deposit: TokenAmount,
     pub pre_commit_epoch: ChainEpoch,
+    /// Integral of active deals over sector lifetime, 0 if CommittedCapacity sector
+    #[serde(with = "bigint_ser")]
+    pub deal_weight: DealWeight,
+    /// Integral of active verified deals over sector lifetime
+    #[serde(with = "bigint_ser")]
+    pub verified_deal_weight: DealWeight,
 }
+
+/// Information stored on-chain for a proven sector.
 #[derive(Debug, PartialEq, Clone, Serialize_tuple, Deserialize_tuple)]
 pub struct SectorOnChainInfo {
-    pub info: SectorPreCommitInfo,
-    /// Epoch at which SectorProveCommit is accepted
-    pub activation_epoch: ChainEpoch,
-    /// Integral of active deals over sector lifetime, 0 if CommittedCapacity sector
+    pub sector_number: SectorNumber,
+    /// The seal proof type implies the PoSt proofs
+    pub seal_proof: RegisteredSealProof,
+    /// CommR
+    pub sealed_cid: Cid,
+    pub deal_ids: Vec<DealID>,
+    /// Epoch during which the sector proof was accepted
+    pub activation: ChainEpoch,
+    /// Epoch during which the sector expires
+    pub expiration: ChainEpoch,
+    /// Integral of active deals over sector lifetime
     #[serde(with = "bigint_ser")]
     pub deal_weight: DealWeight,
     /// Integral of active verified deals over sector lifetime
     #[serde(with = "bigint_ser")]
     pub verified_deal_weight: DealWeight,
+    /// Pledge collected to commit this sector
+    #[serde(with = "bigint_ser")]
+    pub initial_pledge: TokenAmount,
+    /// Expected one day projection of reward for sector computed at activation time
+    #[serde(with = "bigint_ser")]
+    pub expected_day_reward: TokenAmount,
+    /// Expected twenty day projection of reward for sector computed at activation time
+    #[serde(with = "bigint_ser")]
+    pub expected_storage_pledge: TokenAmount,
 }
 
 #[derive(Debug, PartialEq, Clone, Serialize_tuple, Deserialize_tuple)]
 pub struct ChainSectorInfo {
```

Comment:
I don't see this type anywhere, was it just kept because it's used in the state api rn?

Reply:
I'm not sure what you're suggesting? Should this type be somewhere else?

Reply:
I just don't know what it's for, since I don't see an equivalent in specs-actors or Lotus. Fine to just keep as is and can remove once the state API is fixed. Maybe add a TODO to this to make sure it doesn't get ignored?